src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,178 @@
+package duke.parser;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.*;
+import duke.command.*;
+
+import java.io.IOException;
+
+/** Parser to interpret and execute commands. */
+public class Parser {
+
+    /**
+     * Interprets user input and creates the appropriate command.
+     *
+     * @param input    User input during runtime.
+     * @param storage  Storage object to save the task list that is updated in this method.
+     * @param taskList Task list object to store the tasks that are created from user command in this method.
+     * @return An appropriate command.
+     * @throws DukeException If there is an error in parsing user input
+     * @throws IOException   If there is an error in saving task list.
+     */
+    public static Command parseUserInput(String input, Storage storage, TaskList taskList) throws DukeException {
+
+        String[] inputArr = input.split(" ");
+        String instruction = inputArr[0];
+
+        Command command = null;
+
+        switch (instruction) {
+
+        case "done": {
```

Comment:
I don't think curly braces are necessary for [`case` statements](https://nus-cs2103-ay1920s2.github.io/website/coding-standards/java/intermediate.html#layout)....